pymeasure/instruments/common_base.py

```diff
@@ -119,9 +126,14 @@ class CommonBase:
     # Prefix used to store reserved variables
     __reserved_prefix = "___"
 
-    def __init__(self, **kwargs):
+    def __init__(self, preprocess_reply=None, **kwargs):
         self._special_names = self._setup_special_names()
         self._create_channels()
+        if preprocess_reply is not None:
+            warn(("Parameter `preprocess_reply` is deprecated in CommonBase. "
+                 "Implement it in the instrument instead."),
+                 FutureWarning)
```

Comment:
I have to admit I'm confused by the message. IIUC, if one uses a `preprocess_reply` argument in an Instrument initialisation, this gets passed here, so one gets this message, although `preprocess_reply` was used in `Instrument` already, not in `CommonBase` (which an instrument implementor would not touch anyway).
What did you intend "Implement it in the instrument instead." to mean?

Reply:
> Should we mention CommonBase here?

I added it, to distinguish it from the similar message in the Adapter. With the additional comment regarding `read`, it will be distinguishable.

> What did you intend "Implement it in the instrument instead." to mean? Implement the functionality in the instrument's read(), right?

`read` is a hot candidate. You could use `values` as well (depending on the circumstances). I offer `read` as a suggestion, below.

What about this
```suggestion
            warn(("Parameter `preprocess_reply` is deprecated. "
                 "Implement it in the instrument, e.g. in `read`, instead."),
                 FutureWarning)
```
?